script/src/types.rs

```diff
@@ -1012,6 +1012,13 @@ where
     }
 }
 
+/// When the vm is initialized, arguments are loaded onto the stack.
+/// This enum specifies how to locate these arguments.
+pub enum VmArgs {
```

Comment:
Sorry I missed this earlier, please add enough derive macros for this structure(e.g., `Clone Debug PartialEq Eq Hash`). We've add enough structures without derived macros in the past for them to be developer friendly.